Podcast/Cache.swift

```diff
@@ -0,0 +1,107 @@
+        // call. 
+        episodeCache = [:]
+    }
+    
+    // Takes in episode JSON and adds/updates cache
+    // Adds new if not present, updates cache if already present
+    // ONLY should be called from endpoint requests!!!
+    func update(episodeJson: JSON) -> Episode {
+        let id = episodeJson["id"].stringValue
+        if let episode = episodeCache[id] {
+            // Update current episode object to maintain living object
+            episode.title = episodeJson["title"].stringValue
+            episode.descriptionText = episodeJson["summary"].stringValue
+            episode.isRecommended = episodeJson["is_recommended"].boolValue
+            episode.isBookmarked = episodeJson["is_bookmarked"].boolValue
+            episode.numberOfRecommendations = episodeJson["recommendations_count"].intValue
+            episode.seriesTitle = episodeJson["series"]["title"].stringValue
+            episode.seriesID = episodeJson["series"]["id"].stringValue
+            episode.duration = episodeJson["duration"].stringValue
+            episode.tags = episodeJson["tags"].stringValue.components(separatedBy: ";").map({ tag in Tag(name: tag) })
+            episode.audioURL = URL(string: episodeJson["audio_url"].stringValue)
+            episode.dateCreated = DateFormatter.parsingDateFormatter.date(from: episodeJson["pub_date"].stringValue) ?? Date()
+            episode.smallArtworkImageURL = URL(string: episodeJson["series"]["image_url_sm"].stringValue)
+            episode.largeArtworkImageURL = URL(string: episodeJson["series"]["image_url_lg"].stringValue)
+            return episode
+        } else {
+            let episode = Episode(json: episodeJson)
+            episodeCache[id] = episode
+            return episode
+        }
+    }
+    
+    func update(seriesJson: JSON) -> Series {
+        let id = seriesJson["id"].stringValue
+        if let series = seriesCache[id] {
+            series.title = seriesJson["title"].stringValue
+            series.smallArtworkImageURL = URL(string: seriesJson["image_url_sm"].stringValue)
+            series.largeArtworkImageURL = URL(string: seriesJson["image_url_lg"].stringValue)
+            let lastUpdatedString = seriesJson["last_updated"].stringValue
+            series.lastUpdated = DateFormatter.parsingDateFormatter.date(from: lastUpdatedString) ?? Date()
+            series.author = seriesJson["author"].stringValue
+            series.isSubscribed = seriesJson["is_subscribed"].boolValue
+            series.numberOfSubscribers = seriesJson["subscribers_count"].intValue
+            series.tags = seriesJson["genres"].stringValue.components(separatedBy: ";").map({ tag in Tag(name: tag)})
+            return series
+        } else {
+            let series = Series(json: seriesJson)
+            seriesCache[id] = series
+            return series
+        }
+    }
+    
+    func update(userJson: JSON) -> User {
+        let id = userJson["id"].stringValue
+        print(id)
```

Comment:
remove this print statement, have only one `return user`, abstract json logic to user model